src/oatpp-test/web/ClientServerTestRunner.hpp

```diff
@@ -89,21 +89,25 @@ class ClientServerTestRunner {
     std::atomic<bool> running(true);
     std::mutex timeoutMutex;
     std::condition_variable timeoutCondition;
+    bool runConditionForLambda = true;
 
     m_server = std::make_shared<oatpp::network::Server>(m_connectionProvider, m_connectionHandler);
     OATPP_LOGD("\033[1;34mClientServerTestRunner\033[0m", "\033[1;34mRunning server on port %s. Timeout %lld(micro)\033[0m",
                m_connectionProvider->getProperty("port").toString()->c_str(),
                timeout.count());
 
-    std::thread serverThread([this]{
-      m_server->run();
-    });
+    std::function<bool()> condition = [runConditionForLambda](){
```

Comment:
Not meant to be merged in this PR, this is just an dirty and not well engineered example.